_pytest/main.py

```diff
@@ -468,6 +475,11 @@ def _parsearg(self, arg):
         if self.config.option.pyargs:
             parts[0] = self._tryconvertpyarg(parts[0])
         relpath = parts[0].replace("/", os.sep)
+        pattern_line = r':(?P<line>[0-9]+)$'
+        match = re.search(pattern_line, relpath)
+        if match:
+            relpath = ''.join(relpath.split(':')[:-1])
+            self.line = match.group("line")
```

Comment:
You can use the result from `re.search`, e.g `match.start()` and `.end()` instead of the additional `.split()`.

Reply:
Handle `:lineno` should only be done when the path itself does not exist.  `foo:21` is a valid filename on Linux, and there `foo:21:3` would be used to specify the line number - a bit constructed since it would be `test_foo:21.py` then, but still valid.

Reply:
Fixed